src/libs/routes/authmiddleware.ts

```diff
@@ -0,0 +1,23 @@
+import * as jwt from 'jsonwebtoken';
+import hasPermission from '../permission' 
+
+export default (module , permissionType) =>(req , res, next) => {
+    try {
+        console.log('module and permission is ', module , permissionType);
+        console.log('header', req.header('authorization'));
+        const token = req.header('authorization');
+        const decode = jwt.verify(token, 'qwertyuiopasdfghjklzxcvbnm123456');
+        console.log('decoded user', decode);
+        console.log('authorized', hasPermission(module, permissionType, decode.role));
```

Comment:
If the user does not has permission for the module it should not get access and should show a 403 error. You have not added a check for that